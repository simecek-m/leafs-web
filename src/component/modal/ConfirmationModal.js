import React from "react";
import "component/modal/ConfirmationModal.sass";
import { withTranslation } from "react-i18next";
import { withModal } from "modal/withModal";

export function ConfirmationModal({ t, modal = {}, closeModal = () => {} }) {
  return modal.visible ? (
    <div className="confirmation-modal-component" onClick={() => closeModal()}>
      <div className="modal-body" onClick={e => e.stopPropagation()}>
        <h2 className="modal-title">{modal.title}</h2>
        <div className="modal-description">{modal.text}</div>
        <div className="modal-confirmation-button">
          <span
            className="confirmation-text"
            onClick={() => {
              modal.onConfirm();
              closeModal();
            }}
          >
            {t("modal.confirmation")}
          </span>
        </div>
      </div>
    </div>
  ) : null;
}

export default withModal(withTranslation()(ConfirmationModal));
