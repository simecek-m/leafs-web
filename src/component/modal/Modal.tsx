import { AnimatePresence } from "framer-motion";
import { FC, PropsWithChildren } from "react";
import { motion } from "framer-motion";

interface ModalProps extends PropsWithChildren {
  isOpen: boolean;
  onClose: () => void;
}

const Modal: FC<ModalProps> = ({ isOpen, onClose, children }) => {
  return (
    <AnimatePresence>
      {isOpen && (
        <motion.div
          initial={{ opacity: 0 }}
          animate={{ opacity: 1 }}
          exit={{ opacity: 0 }}
          className="absolute top-0 left-0 z-40 flex h-screen w-screen items-center justify-center bg-black bg-opacity-70"
          onClick={onClose}
        >
          <motion.div
            initial={{ y: -100 }}
            animate={{ y: 0 }}
            exit={{ y: -100 }}
            className="m-2 flex w-full max-w-3xl flex-col rounded-3xl bg-white p-6 dark:bg-gray-800 md:w-1/2"
            onClick={(e) => e.stopPropagation()}
          >
            {children}
          </motion.div>
        </motion.div>
      )}
    </AnimatePresence>
  );
};

export default Modal;
