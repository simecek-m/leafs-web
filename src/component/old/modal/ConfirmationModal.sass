@import "colors.sass"

.confirmation-modal-component
  background-color: $transparent-dark
  position: fixed
  top: 0
  left: 0
  bottom: 0
  right: 0
  width: 100vw
  height: 100vh
  z-index: 10
  text-align: left

  .modal-body
    width: 70%
    max-width: 640px
    min-height: 200px
    background-color: $text-primary
    margin: 120px auto
    box-shadow: 0 0 20px $black
    border-radius: 30px
    padding: 20px 60px
    color: $background-primary

    .modal-title
      margin: 20px 0
      font-weight: 900
      font-size: 35px

    .modal-description
      font-size: 22px
      font-style: italic
      margin-bottom: 30px

    .modal-confirmation-button
      color: $red
      font-size: 28px
      font-weight: 800
      text-align: right

      .confirmation-text
        cursor: pointer
