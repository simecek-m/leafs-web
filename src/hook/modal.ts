import { useState } from "react";

export const useModal = () => {
  const [isOpen, setOpen] = useState<boolean>(false);
  return { isOpen, onClose: () => setOpen(false), onOpen: () => setOpen(true) };
};
