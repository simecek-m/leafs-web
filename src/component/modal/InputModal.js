import React from "react";
import { withTranslation } from "react-i18next";
import { withModal } from "modal/withModal";
import "component/modal/InputModal.sass";

export function InputModal({ t, modal = {}, closeModal }) {
  const modalInput = React.createRef();
  return modal.visible ? (
    <div className="input-modal-component" onClick={() => closeModal()}>
      <div className="modal-body" onClick={e => e.stopPropagation()}>
        <h2 className="modal-title">{modal.title}</h2>
        <div className="modal-description">{modal.text}</div>
        <input ref={modalInput} className="modal-input" autoFocus />
        <div className="modal-confirmation-button">
          <span
            className="confirmation-text"
            onClick={() => {
              modal.onConfirm(modalInput.current.value);
              closeModal();
            }}
          >
            {t("modal.confirmation")}
          </span>
        </div>
      </div>
    </div>
  ) : null;
}

export default withModal(withTranslation()(InputModal));
