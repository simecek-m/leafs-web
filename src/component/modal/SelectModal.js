import React from "react";
import { withModal } from "modal/withModal";
import SelectItem from "component/modal/component/SelectItem";
import "component/modal/SelectModal.sass";

export function SelectModal({ modal = {}, closeModal = () => {} }) {
  const options = modal.options
    ? modal.options.map((option, index) => (
        <SelectItem
          key={index}
          id={`select-item-${index}`}
          image={option.image}
          text={option.text}
          onClick={() => {
            if (option.onClick) {
              option.onClick();
              closeModal();
            }
          }}
        />
      ))
    : null;
  return modal.visible ? (
    <div className="select-modal-component" onClick={() => closeModal()}>
      <div className="modal-body" onClick={e => e.stopPropagation()}>
        <h2 className="modal-title">{modal.title}</h2>
        <div className="modal-description">{modal.text}</div>
        {options}
      </div>
    </div>
  ) : null;
}

export default withModal(SelectModal);
