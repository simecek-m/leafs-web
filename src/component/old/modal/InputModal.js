import React from "react";
import { withTranslation } from "react-i18next";
import { withModal } from "modal/withModal";
import "component/old/modal/InputModal.sass";

export class InputModal extends React.Component {
  state = {
    value: ""
  };

  submitForm(event) {
    const { modal, closeModal } = this.props;
    event.preventDefault();
    const data = this.state.value;
    modal.onConfirm(data);
    closeModal();
  }

  changeValue(event) {
    const newValue = event.target.value;
    this.setState({ value: newValue });
  }

  render() {
    const { modal = {}, closeModal, t } = this.props;
    return modal.visible ? (
      <div className="input-modal-component" onClick={() => closeModal()}>
        <div className="modal-body" onClick={e => e.stopPropagation()}>
          <h2 className="modal-title">{modal.title}</h2>
          <div className="modal-description">{modal.text}</div>
          <form onSubmit={e => this.submitForm(e)}>
            <input
              name="value"
              value={this.state.value}
              onChange={e => this.changeValue(e)}
              type="text"
              className="modal-input"
              autoFocus
            />
            <button className="modal-confirmation-button" type="submit">
              {t("modal.confirmation")}
            </button>
          </form>
        </div>
      </div>
    ) : null;
  }
}

export default withModal(withTranslation()(InputModal));
