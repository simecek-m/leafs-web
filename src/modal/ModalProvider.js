import React from "react";
import ConfirmationModal from "component/old/modal/ConfirmationModal";
import SelectModal from "component/old/modal/SelectModal";
import InputModal from "component/old/modal/InputModal";
import { CONFIRMATION, SELECT, INPUT } from "modal/types";
import { validateModalData } from "modal/validator";

export const ModalContext = React.createContext();

class ModalProvider extends React.Component {
  constructor(props) {
    super(props);
    const { visible = false, type = CONFIRMATION, ...rest } = this.props.init
      ? this.props.init
      : {};
    this.state = {
      visible,
      type,
      ...rest
    };
  }

  openModal = modal => {
    const modalData = validateModalData(modal);
    this.setState(modalData);
  };

  closeModal = () => {
    this.setState({
      visible: false
    });
  };

  mountModalComponent() {
    switch (this.state.type) {
      case SELECT:
        return <SelectModal />;
      case INPUT:
        return <InputModal />;
      default:
        return <ConfirmationModal />;
    }
  }

  render() {
    return (
      <ModalContext.Provider
        value={{
          modal: this.state,
          openModal: this.openModal,
          closeModal: this.closeModal
        }}
      >
        {this.mountModalComponent()}
        {this.props.children}
      </ModalContext.Provider>
    );
  }
}

export default ModalProvider;
