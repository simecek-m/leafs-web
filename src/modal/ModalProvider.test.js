import React from "react";
import ModalProvider from "modal/ModalProvider";
import { shallow } from "enzyme";
import { CONFIRMATION, INPUT, SELECT } from "modal/types";

const INITIAL_STATE = {
  visible: false,
  type: CONFIRMATION
};

const TEST_CONFIRMATION_MODAL = {
  visible: true,
  type: CONFIRMATION,
  title: "Confirmation Modal",
  text: "Description Modal",
  onConfirm: () => {}
};

const TEST_INPUT_MODAL = {
  visible: true,
  type: INPUT,
  title: "Input Modal",
  text: "Description Modal",
  onConfirm: () => {}
};

const TEST_SELECT_MODAL = {
  visible: true,
  type: SELECT,
  title: "Select Modal",
  text: "Description Modal",
  options: []
};

describe("ModalProvider", () => {
  test("should render provider", () => {
    const wrapper = shallow(<ModalProvider />);
    const state = wrapper.state();
    expect(state).toEqual(INITIAL_STATE);
    expect(wrapper).toMatchSnapshot();
  });

  test("should switch to Confirmation Modal", () => {
    const wrapper = shallow(<ModalProvider />);
    wrapper.instance().openModal(TEST_CONFIRMATION_MODAL);
    const state = wrapper.state();
    expect(state).toEqual(TEST_CONFIRMATION_MODAL);
  });

  test("should switch to Input Modal", () => {
    const wrapper = shallow(<ModalProvider />);
    wrapper.instance().openModal(TEST_INPUT_MODAL);
    const state = wrapper.state();
    expect(state).toEqual(TEST_INPUT_MODAL);
  });

  test("should switch to Select Modal", () => {
    const wrapper = shallow(<ModalProvider />);
    wrapper.instance().openModal(TEST_SELECT_MODAL);
    const state = wrapper.state();
    expect(state).toEqual(TEST_SELECT_MODAL);
  });

  test("should close modal", () => {
    const wrapper = shallow(<ModalProvider />);
    wrapper.instance().openModal(TEST_CONFIRMATION_MODAL);
    let state = wrapper.state();
    expect(state).toEqual(TEST_CONFIRMATION_MODAL);
    wrapper.instance().closeModal();
    state = wrapper.state();
    expect(state.visible).toBeFalsy();
  });
});
