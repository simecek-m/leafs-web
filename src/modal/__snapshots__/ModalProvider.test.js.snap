// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalProvider should render provider 1`] = `
<ContextProvider
  value={
    Object {
      "closeModal": [Function],
      "modal": Object {
        "type": "confirmation",
        "visible": false,
      },
      "openModal": [Function],
    }
  }
>
  <Wrapper />
</ContextProvider>
`;
